src/routes/animal_categories.js

```diff
@@ -0,0 +1,43 @@
+const express = require('express');
+
+const authMiddleware = require('../middlewares/auth');
+
+const AnimalCategoriesService = require('../services/animal_categories');
+
+const AnimalPhotosService = require('../services/animal_photos');
+
+const router = express.Router();
+
+/* Get all the categories */
+router.get('/', async (req, res) => {
+  res.send(await AnimalCategoriesService.readAll());
+});
+
+/* Create new categories with photos */
+router.post('/', authMiddleware, async (req, res) => {
+  const payload = req.body;
+  Object.keys(payload).forEach(async (categoryName) => {
+    if (payload[categoryName] && payload[categoryName].length) {
+      const category = await AnimalCategoriesService.create(categoryName);
+      payload[categoryName].forEach(async (photoUrl) => {
+        await AnimalPhotosService.create({ category_id: category.id, photo_url: photoUrl });
+      });
+    }
+  });
+  res.status(200).send('Posted successfully!');
+});
+
+/** Delete new category and cascades the related photos as well. */
+router.delete('/:id', authMiddleware, async (req, res) => {
+  if (req.params.id) {
+    try {
+      const category = await AnimalCategoriesService.delete(req.params.id);
+      res.send(category);
+    } catch (err) {
+      // log the error somewhere.
+      res.send([]);
+    }
+  }
```

Comment:
try/catch error handling appears to be inconsistent across routes, resulting in errors such as hanging responses. How would you handle errors for your routes?

Reply:
yes it’s inconsistent across routes because it’s not handled by error reporting middleware. I would create a middleware so that when error is passed into the next function express knows that error is there and we can use that middleware to control error reporting/logging as well.